fix build on MacOS

garymm opened the pull request:
fix build on MacOS

- Fix zmqpp build:
  When using system zmqpp, previously the wrong variable was being set.
  When building from submodule, previously it was using make in a way that only worked if the compiler was g++. Now we use CMake which works on MacOS.

- In client: don't assume `high_resolution_clock` has `to_time_t`.

Note:
I didn't get the headless support building on MacOS yet. This works with -DBUILD_HEADLESS=FALSE.

This PR is Ready for Review.

## How to test

Build the code.

garymm committed:
Fix zmqpp build:
* When using system zmqpp, previously the wrong variable was being set.
* When building from submodule, previously it was using make in a way
  that only worked if the compiler was g++. Now we use CMake which works
  on MacOS.

garymm committed:
fix build on MacOS

see https://stackoverflow.com/a/32556992/1720894

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/EleutherAI/minetest?pullRequest=81) <br/>All committers have signed the CLA.

neverix commented:
The Makefile uses `util/minetester/build_minetest.sh`, which in turn uses `nproc`. I haven't finished building, but that should probably be changed

garymm commented:
I should clarify, I didn't get the headless support building on MacOS yet. This works with `-DBUILD_HEADLESS=FALSE`.

AI-WAIFU reviewed (changes requested):
Can you add a check and raise an error when building for MacOS with -DBUILD_HEADLESS=TRUE? Just so fail silently/inexplicably.

garymm committed:
Fail if trying to build headless on MacOS.

AFAICT this is doomed to fail anyways, so failing early with an
informative error message is an improvement.

It should be possible to get it working though.

AI-WAIFU reviewed (approved):
LGTM
